### ISSUE TITLE
cuda test failed

### ISSUE THREAD

**Issue by beew** [hypre-space/hypre#395]
I compiled hypre-2.21.0 with

```
/configure --prefix=$HOME/opt/hypre --with-cuda --enable-shared --enable-unified-memory 

```
compilation was successful but make check produced

```
CUDA ERROR (code = 98, invalid device function) at hypre_general.c:151
Aborted (core dumped)
-rw-rw-r-- 1 bernard bernard 1568 Jun  7 14:10 ij.err
-rw-rw-r-- 1 bernard bernard 1859 Jun  7 14:10 sstruct.err
-rw-rw-r-- 1 bernard bernard    0 Jun  7 14:10 struct.err
```

OS Ubuntu 20.04, Cuda 11.0 Nvidia-GTX 1070

**Comment by beew** [comment]
Ok, figured it out.

Apparently the failure was caused by hypre not detecting cuda-arch correctly.

My gpu is GTX1070 (Pascal) so cuda-arch should be

-gencode arch=compute_61,code=sm_61

instead of

-gencode arch=compute_70,code=sm_70

So I manually changed the following line in config/Makefile.config after running ./configure

```
CUDA_ARCH          = -gencode arch=compute_70,code=sm_70
```
to

```
CUDA_ARCH          = -gencode arch=compute_61,code=sm_61

```

This fixed the problem, no more error.

So is there a way to supply this information during ./configure?
